index.js

```diff
@@ -209,12 +222,18 @@ function compression (options) {
       })
 
       stream.on('end', function onStreamEnd () {
-        _end.call(res)
+        endOnce.call(res)
       })
 
       _on.call(res, 'drain', function onResponseDrain () {
         stream.resume()
       })
+
+      onFinished(res, function onFinished () {
+        if (ended) {
+          endOnce.call(res)
+        }
+      })
```

Comment:
This addition takes care of situations where a response becomes "finished" soon after `res.end()` gets called by a route handler. Without this code, if the events occur close enough to one another, the stream may (at least in principle) get stuck in a "paused" state, never emitting the `end` event that would normally invoke the original `res.end` function.